File: Python/LinearSearchArray/LinearSearch.py
class Solution:
  def LinearSearch(target, data):
    length = len(data)
    for x in range(length):
      if data[x] == target:
        return x
    return -1
    
"""
parameter (target, data)
* target = value to searching index
* data   = list of data (string, integer, etc.)
"""
print(Solution.LinearSearch(3, [1, 2, 3, 5, 4]))  # index 2
print(Solution.LinearSearch(4, [1, 2, 3, 5, 4]))  # index 4
print(Solution.LinearSearch(1, [1, 2, 3, 5, 4]))  # index 0
print(Solution.LinearSearch(7, [1, 2, 3, 5, 4]))  # -1 : not found

print(Solution.LinearSearch("Amelia", ["John", "Susan", "Amelia", "Robert"])) # index 2 (third index)

File: NodeJS/LinearSearchArray/LinearSearch.js
/**
 * @description   Search number of index from array
 * @param         target  {int}
 * @param         data    {array}
 * @returns       index
 */
const LinearSearch = (target, data) => {
  for ( let i=0; i < data.length; i++ )
    if ( data[i] == target ) return i;
  return -1;
}

console.log(LinearSearch(3, [23, 44, 24, 69, 200]));    // -1  
console.log(LinearSearch(44, [23, 44, 24, 69, 200]));   // 1
console.log(LinearSearch(100, [23, 44, 24, 69, 200]));  // -1
console.log(LinearSearch(69, [23, 44, 24, 69, 200]));   // 3